src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -116,94 +116,82 @@ open class Singleton<T : Entity<T>> constructor(val entityCtor: () -> T) : Handl
         entity = entityCtor.invoke().decodeEntity(added)
     }
 
-    fun get(): T? {
-        return entity
-    }
+    fun get() = entity
 
     fun set(entity: T) {
         this.entity = entity
         val encoded = entity.encodeEntity()
         singletonSet(
-            this.particle!!.toWasmAddress(),
+            this.particle.toWasmAddress(),
             this.toWasmAddress(),
             encoded.toWasmString()
         )
     }
 
     fun clear() {
         entity = entityCtor()
-        singletonClear(this.particle!!.toWasmAddress(), this.toWasmAddress())
+        singletonClear(this.particle.toWasmAddress(), this.toWasmAddress())
     }
 }
 
 class Collection<T : Entity<T>> constructor(private val entityCtor: () -> T) : Handle(),
     Iterable<T> {
-    override fun iterator(): Iterator<T> {
-        return entities.values.iterator()
-    }
 
     private val entities: MutableMap<String, T> = mutableMapOf()
 
+    override fun iterator() = entities.values.iterator()
+
     override fun sync(encoded: String) {
         entities.clear()
         add(encoded)
     }
 
     override fun update(added: String, removed: String) {
         add(added)
-        val decoder = StringDecoder(removed)
-            var num = decoder.getInt(":")
+        with (StringDecoder(removed)) {
+            var num = getInt(":")
             while (num-- > 0) {
-                val len = decoder.getInt(":")
-                val chunk = decoder.chomp(len)
+                val len = getInt(":")
+                val chunk = chomp(len)
                 // TODO: just get the id, no need to decode the full entity
                 val entity = entityCtor.invoke().decodeEntity(chunk)!!
                 entities.remove(entity.internalId)
             }
+        }
     }
 
-    fun size(): Int {
-        return entities.size
-    }
-
-    fun empty(): Boolean {
-        return entities.isEmpty()
-    }
+    fun size() = entities.size
```

Comment:
Size could be a property:

```kotlin
val size: Int
  get() = entities.size
```

Reply:
Is there a benefit to that?

Reply:
It's more idiomatic/conventional for kotlin.

Reply:
Done